tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -504,75 +551,43 @@ ROOT::NTupleSize_t ROOT::Experimental::RNTupleJoinProcessor::LoadEntry(ROOT::NTu
    if (entryNumber == kInvalidNTupleIndex)
       fEntry->Reset();
 
-   if (entryNumber >= fPageSource->GetNEntries())
-      return ROOT::kInvalidNTupleIndex;
-
-   // Read the values of the primary ntuple. If no join table is used (i.e., the join is aligned), also read the values
-   // of auxiliary ntuples.
-   for (const auto &[_, fieldContext] : fFieldContexts) {
-      if (!fieldContext.IsAuxiliary() || !HasJoinTable()) {
-         auto &value = fEntry->GetValue(fieldContext.fToken);
-         value.Read(entryNumber);
-      }
-   }
+   if (fPrimaryProcessor->LoadEntry(entryNumber) == kInvalidNTupleIndex)
+      return kInvalidNTupleIndex;
 
    fCurrentEntryNumber = entryNumber;
    fNEntriesProcessed++;
 
-   // If no join table is used (i.e., the join is aligned), there's nothing left to do.
-   if (!HasJoinTable())
-      return entryNumber;
+   if (!HasJoinTable()) {
+      for (auto &auxProcessor : fAuxiliaryProcessors) {
+         // TODO handle aux processor with fewer entries
+         auxProcessor->LoadEntry(entryNumber);
+      }
+   }
 
-   // First build the join tables if this hasn't been done yet.
    if (!fJoinTablesAreBuilt) {
       for (unsigned i = 0; i < fJoinTables.size(); ++i) {
-         fJoinTables[i]->Add(*fAuxiliaryPageSources[i]);
+         fAuxiliaryProcessors[i]->AddEntriesToJoinTable(*fJoinTables[i]);
       }
-
       fJoinTablesAreBuilt = true;
    }
 
-   // Collect the values of the join fields for this entry.
-   std::vector<void *> valPtrs;
-   valPtrs.reserve(fJoinFieldTokens.size());
-   for (const auto &token : fJoinFieldTokens) {
-      auto ptr = fEntry->GetPtr<void>(token);
-      valPtrs.push_back(ptr.get());
-   }
```

Comment:
I believe we will need this back to deal with changing pointers in the entry (see other comments)